src/hypertable.c

```diff
@@ -2551,6 +2552,30 @@ ts_hypertable_create_compressed(Oid table_relid, int32 hypertable_id)
 	Relation rel;
 
 	rel = table_open(table_relid, AccessExclusiveLock);
+	int32 row_size = MAXALIGN(SizeofHeapTupleHeader);
+	/* estimate tuple width of compressed hypertable */
+	row_size += get_relation_data_width(table_relid, NULL);
+	/* assuming data gets compressed to half of its original size */
+	if ((row_size / 2) > MaxHeapTupleSize)
```

Comment:
Why did you pick a compression ratio of 50%?

Reply:
This calculation is not correct since a toasted pointer consumes 18 bytes. 

https://www.postgresql.org/docs/current/storage-toast.html

Reply:
> This calculation is not correct since a TOAST pointer consumes 18 bytes.
> 
> https://www.postgresql.org/docs/current/storage-toast.html

Hmm... so this means that if you assume that all columns smaller than 18 bytes are toasted, you would be safe.

Basically, something like `columns.map(|x| max(x.size, 18)).sum() > MaxHeapTupleSize`.

Reply:
It's more like `sum(if col.varlen then 18 else col.size end) > MaxHeapTupleSize`
Also get_relation_data_width since not the right function to use here as you want the theoretical limits of the tuple not which values are currently stored if any.

Reply:
But since this is checking the size of the column of the _uncompressed_ table, which could be a fixed size column that is smaller than 18 bytes (e.g., an integer), it would mean that once the column is compressed, it will turn into an array, which will then (hopefully) be toasted and will then occupy 18 bytes.

Wouldn't this check fail for a table with, say, 600 integer columns (which would occupy 600 * 8 = 4800 bytes in the uncompressed table and 600 * 18 = 10800 in the compressed table)?

Reply:
it should be checking the compressed table, otherwise it has to take segment by and order by into account

Reply:
If we do the check on the compressed table, then the check would be accurate, but unless I am misreading it, the check right now is on the uncompressed table, so then the calculations are off.

We can still make a conservative estimate by assuming that columns in the uncompressed table take at least 18 bytes, and that we have the additional columns, which would account also for segmentby columns that have a size that is less than 18 bytes, and hence be stored with the "uncompressed" size.

Not sure how the orderby would affect the row size.